includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -625,12 +625,12 @@ static function ( $validation_error ) {
 		// Remove the element if it is has an invalid layout.
 		$layout_validity = $this->is_valid_layout( $tag_spec, $node );
 		if ( true !== $layout_validity ) {
-			$this->remove_invalid_child( $node, $layout_validity );
-			return null;
+			$sanitized = $this->remove_invalid_child( $node, $layout_validity );
+			return $sanitized ? null : $this->get_required_script_components( $node, $tag_spec, $attr_spec_list );
```

Comment:
Optional thought:
If the return value types of `process_node()`, `remove_invalid_child()` and `get_required_script_components()` are adapted so they have the same understanding of no operation (either all `null` or all `false`), then code like this could be written in a more readable way:
```suggestion
			return $this->remove_invalid_child( $node, $layout_validity )
				|| $this->get_required_script_components( $node, $tag_spec, $attr_spec_list );
```
This reads like "_either_ remove the node _or_ get its script components".